### PR TITLE
Final Capstone Project: Rent-Eaze Booking System - Merge Request

### PR DESCRIPTION
**Pull request by lRebornsl**
**Overview**
This pull request is a formal request to merge our development team's cumulative and concerted efforts from the `dev` branch into the `main` branch. Our Final Capstone Project is a sophisticated rental house booking system, designed and tailored following the prototype provided by Murat Korkmaz on Behance, with our unique spin - instead of motorcycles, we facilitate the booking of rental houses.

Our development team, composed of Javier Grau, Manuel Sanchez, and Anthony Vásquez, has meticulously adhered to coding best practices, proper Gitflow, and comprehensive professional documentation throughout the project lifecycle.

[Pull Request from front-end](https://github.com/grauJavier/final-capstone-front-end/pull/52)

**General Requirements Met**
- Zero linter errors across the codebase.
- Strict adherence to Gitflow practices.
- Professional-level documentation and code commenting.

**Project Implementation Details**

**Chosen Theme**
- Our application is built around the concept of booking rental houses, aiming to streamline the process of scheduling a house with an intuitive user interface.

**Core Features Implemented**
- Normal user login system by typing the email and password.
- The navigation panel provides seamless access to:
  - List of houses.
  - House reservation form.
  - Functionality to add new houses to the system.
  - Option to remove their own houses from the system.
- The main page features a listing of houses available for users.
- Detail pages for each house, with an option for users to reserve it.
- A fully responsive design, with tailored user experiences for both mobile and desktop versions.

**Additional Features (Beyond Basic Requirements)**
- The reservation process is enhanced by allowing users to select only a future date.

**Comprehensive Documentation**
- Detailed API documentation has been included to facilitate understanding and potentially scaling the application in the future.

**Useful Links**
- Back-End Documentation: https://renteaze-d1cc8b293660.herokuapp.com/api-docs/index.html
- Full project deployment: https://rent-eaze.onrender.com/

**Merge Request Justification**
With all required features meticulously implemented, we are confident that our project meets the specified criteria. We kindly request the review and approval for the merging of our `dev` branch into `main`.

We eagerly await any feedback and are fully prepared to conduct further modifications if recommended by our esteemed reviewers.

Thank you for your consideration and time.

Warm regards,
@grauJavier 
@Luffytaro22 
@lRebornsl

### CODE REVIEW

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @lRebornsl,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  All tests but 1 are passing✔️
-  Nice code organization ✔️ 
-  API is working well✔️ 
-  Good documentation✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md [hunk 1]
```diff
@@ -1 +1,270 @@
...
+
+```
+
+After you've installed RSpec, you can run the tests with:
+
+```sh
+
+rspec
+
+```
```
[the remark this answers is not in the document]
Comment (Luffytaro22) [reply README.md:182]: Hello!
Thanks for your feedback 🙌🏼
This is solved 😄

File: app/controllers/places_controller.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    else
+      render json: place.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    place = @places.find(params[:id])
+    place.destroy
+    head :no_content
+  end
```
[the remark this answers is not in the document]
Comment (Luffytaro22) [reply app/controllers/places_controller.rb]: Hello!
Thanks for your feedback 🙌🏼
This is solved 😄

File: app/controllers/reservations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      render json: { errors: reservation.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE
+  def destroy
+    reservation = @reservations.find(params[:id])
+    reservation.destroy
+    head :no_content
+  end
```
[the remark this answers is not in the document]
Comment (Luffytaro22) [reply app/controllers/reservations_controller.rb]: Hello!
Thanks for your feedback 🙌🏼
This is solved 😄

File: README.md [hunk 4]
```diff
@@ -1 +1,270 @@
...
+### Database Setup
+
+Create and migrate your database with:
+
+```sh
+
+rails db:create
+rails db:migrate
+
+```
```
[the remark this answers is not in the document]
Comment (lRebornsl) [reply README.md]: Hello!
Thanks for your feedback 🙌🏼
This is solved 😄

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @lRebornsl,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!










#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md [hunk 5]
```diff
@@ -1 +1,271 @@
...
+- gem install rails
+- bundle install
+
+### Database Setup
+
+Create and migrate your database with:
+
+```sh
+
+rails db:create
```
[the remark this answers is not in the document]
Comment (Luffytaro22) [reply README.md:155]: Hello!
Thanks for your feedback 🙌🏼
This is solved 😄

**Review by Meltrust — APPROVED** [review]
#### Hi @lRebornsl,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Readme instructions have been improved✔️ 
- All tests are now passing✔️ 
- Great job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**